pkg/cloudproviders/k8s/k8s.go

```diff
@@ -0,0 +1,348 @@
+package k8s
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/spf13/viper"
+
+	"github.com/atlassian/gostatsd"
+	"github.com/atlassian/gostatsd/pkg/util"
+
+	"github.com/sirupsen/logrus"
+	core_v1 "k8s.io/api/core/v1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	core_v1inf "k8s.io/client-go/informers/core/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/flowcontrol"
+)
+
+// NeverMatchRegex is a regex string that will never match anything
+const NeverMatchRegex = "k^"
+
+var (
+	// DefaultAnnotationTagRegex is the default ParamAnnotationTagRegex. Everything beginning with a standard
+	// gostatsd string is included by default. The default sets the tag name as the part after the "/", for example
+	// gostatsd.atlassian.com/tag1 would return "tag1" as the tag name after matching.
+	DefaultAnnotationTagRegex = "^" + regexp.QuoteMeta(AnnotationPrefix) + DefaultTagCaptureRegex
+	// DefaultLabelTagRegex is the default ParamLabelTagRegex. Every label is ignored by default.
+	DefaultLabelTagRegex = NeverMatchRegex // regex will never match anything
+)
+
+const (
+	// ProviderName is the name of the k8s cloud provider.
+	ProviderName = "k8s"
+	// PodsByIPIndexName is the name of the index function storing pods by IP.
+	PodsByIPIndexName = "PodByIP"
+	// AnnotationPrefix is the annotation prefix that is turned into tags by default.
+	AnnotationPrefix = "gostatsd.atlassian.com/"
+	// DefaultTagCaptureRegex is a regex subexpression to capture from now until the end of the regex as the tag name.
+	DefaultTagCaptureRegex = "(?P<tag>.*)$"
```

Comment:
I suggest to make `tag` a constant to avoid hardcoding it in more than one place.

Reply:
If you end it with `$`, I'd suggest to also start with `^` to make the intent clearer.

Reply:
The intent was really just to be a reusable thing to capture a tag at any point inside a regex. I split it out from the main regex for documentation purposes really. I will remove the $ and just add that into the combination though to make it more generic.